### PR TITLE
Added Instrument lookup

### PR DESCRIPTION
**Pull request by thmiceli**
I don't have working examples for Curve and Government lookup requests yet but I should be able to come up with some. 
There isn't an example for FieldSearch either. I added the service to the map because Yu Ping mentioned it.
We should probably try to have at least one example for each service that is supported.

### CODE REVIEW

File: lib/blpapi-wrapper.ts [hunk 1]
```diff
@@ -71,15 +71,25 @@ var EVENT_TYPE = {
   , PARTIAL_RESPONSE: 'PARTIAL_RESPONSE'
 };
 
+// Map of HTTP request names to the BLAPI request and response names.
```
Comment (ericlu88) [inline lib/blpapi-wrapper.ts:74]: BLPAPI

File: lib/blpapi-wrapper.ts [hunk 2]
```diff
@@ -71,15 +71,25 @@ var EVENT_TYPE = {
...
+var NAME_MAP: { [s: string]: { reqName: string; resName: string; } } = {
```
Comment (yupinghu) [inline lib/blpapi-wrapper.ts:75]: This is a good test case for the style guide. :) @ericlu88 @ericvw what do you think of the formatting for NAME_MAP's value? It looks OK but it technically violates the current version of the style guide.

Also, we may want to consider a section on how to format type definitions.

File: lib/blpapi-wrapper.ts [hunk 3]
```diff
@@ -71,15 +71,25 @@ var EVENT_TYPE = {
...
+    'FieldInfo':        { reqName: 'FieldInfoRequest',
+                          resName: 'fieldResponse' },
+    'FieldSearch':      { reqName: 'FieldInfoRequest',
+                          resName: 'fieldResponse' },
+    'InstrumentLookup': { reqName: 'instrumentListRequest',
+                          resName: 'InstrumentListResponse' },
+    'CurveLookup':      { reqName: 'curveListRequest',
```
Comment (yupinghu) [inline lib/blpapi-wrapper.ts:82]: Do we want this to be alphabetical?

File: lib/blpapi-wrapper.ts [hunk 4]
```diff
@@ -71,15 +71,25 @@ var EVENT_TYPE = {
...
+                          resName: 'CurveListResponse' },
+    'GovernmentLookup': { reqName: 'govtListRequest',
+                          resName: 'GovtListResponse' }
+};
+
```
Comment (ericlu88) [inline lib/blpapi-wrapper.ts:87]: Seems to missing `HistoricalData` & `ReferenceData` in this list(The other two I am aware of). I am not sure if there's a complete list of request names in the api doc though.

File: lib/blpapi-wrapper.ts [hunk 5]
```diff
@@ -71,15 +71,25 @@ var EVENT_TYPE = {
...
+                          resName: 'fieldResponse' },
+    'FieldSearch':      { reqName: 'FieldInfoRequest',
+                          resName: 'fieldResponse' },
+    'InstrumentLookup': { reqName: 'instrumentListRequest',
+                          resName: 'InstrumentListResponse' },
+    'CurveLookup':      { reqName: 'curveListRequest',
+                          resName: 'CurveListResponse' },
+    'GovernmentLookup': { reqName: 'govtListRequest',
+                          resName: 'GovtListResponse' }
+};
```
Comment (yupinghu) [inline lib/blpapi-wrapper.ts:86]: Need to add entries for //blp/refdata requests. I believe HistoricalData and ReferenceData are the only ones we generally have tested, although there's no real reason the rest shouldn't work. The reqName and resName should just be FooRequest and FooResponse for all refdata requests.

Comment (thmiceli) [inline lib/blpapi-wrapper.ts:86]: Yes it looks like I missed these two services.

File: lib/blpapi-wrapper.ts [hunk 6]
```diff
@@ -71,15 +71,25 @@ var EVENT_TYPE = {
   , PARTIAL_RESPONSE: 'PARTIAL_RESPONSE'
 };
 
+// Map of HTTP request names to the BLAPI request and response names.
+var NAME_MAP: { [s: string]: { reqName: string; resName: string; } } = {
+    'FieldInfo':        { reqName: 'FieldInfoRequest',
+                          resName: 'fieldResponse' },
+    'FieldSearch':      { reqName: 'FieldInfoRequest',
```
Comment (ericlu88) [inline lib/blpapi-wrapper.ts:78]: `FieldSearchRequest`

**Comment by thmiceli** [comment]
Replaced with PR #74

**Pull request closed without merging**